modxcms/revolution
# Refactor of template-picker

Only the refactoring and a few extra features is my work, everything else is credited to the original author.

### What does it do?
This is a refactor of the template picker feature by @Oetzie 

![Screenshot_2021-03-16 Template BaseTemplate MODX Revolution(1)](https://user-images.githubusercontent.com/5160368/111286005-b07d4b00-867c-11eb-9a1f-5d151f8c8fa0.png)


### Why is it needed?
The original PR #13905 is dated due to the core code base being refactored.

### How to test
**Requires assets to be rebuilt
**Requires an added column `preview_file` added to the modTemplateVar `modx_site_tmplvars` table.
**Requires system setting `enable_template_picker_in_tree` (create as a yes/no boolean)

1. First edit the template in the manager, set a media source for the template and select an image to use.
2. Make sure the system setting `enable_template_picker_in_tree` is set to Yes. (create setting if not there)
2. Click on create a new resource in the resource tree. 
3. Try using the template picker when it appears.


![Screenshot_2021-03-12 Template BaseTemplate MODX Revolution](https://user-images.githubusercontent.com/5160368/110907486-6fb5c700-8348-11eb-8303-da3276755e34.png)


In the original PR there was some modification of the parent resource picker which was applied to the settings tab of the resource edit screen. This was causing display issues of the forms so I removed it. (Unsure of what the intention behind it was)



### Related issue(s)/PR(s)
PR #13905
